### PR TITLE
Delete test data

### PR DESCRIPTION
**Pull request by johnhosolirius**
These fixes ensure that test data is deleted from after cucumber test runs.

The important change is to src/aat/resources/application.yaml which now points to Postgres instead of H2.

The rest of the changes are really just cleaning up the testing "clean up" code

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java [hunk 1]
```diff
@@ -77,7 +98,8 @@ public void an_existing_question_with_id() throws IOException {
         response = restTemplate.exchange(baseUrl + endpoints.get("question"), HttpMethod.POST, request, String.class);
         String json = response.getBody();
         question = (Question) JsonUtils.toObjectFromJson(json, Question.class);
-        this.questionId = question.getQuestionId();
+        this.currentQuestionId = question.getQuestionId();
+        questionIds.add(question.getQuestionId());
     }
 
     @Given("^a standard answer$")
```
Comment (Bishwin) [inline src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java]: reason for creating an instance of the jsonutils class when it only has static methods?

File: src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java [hunk 2]
```diff
@@ -1,71 +1,92 @@
...
-    @Given("^a valid question$")
     /**
      * Creates a question to be used for testing with an answer
      */
-    public void an_existing_question_with_id() throws IOException {
+    @Given("^a valid question$")
+    public void an_existing_question() throws IOException {
         Question question = new Question();
         question.setSubject("foo");
         question.setQuestionText("question text");
```
Comment (phillipDenness) [inline src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java]: If the postgres DB has a foreign key constraint on the question 'online_hearing_id', then the test fails(providing there is no online hearing with online_hearing_id = 1. Therefore I think you need to create a online_hearing with id =1 as a prerequisite for the question entity.

File: src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java (hunk 1 above)
Comment (johnhosolirius) [reply src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java]: It's a utility class and re-usable from anywhere. There's no need to classes to create their own instance of JsonUtils.

File: src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java (hunk 2 above)
Comment (johnhosolirius) [reply src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java]: At this point I don't think there is actually an FK on Online Hearing cause "someone" never put it in and the online hearing/question wasn't ready. That's why the test passes. 
I'm not sure how to handle this. Let's discuss it in the standup

**Pull request merged**